### PR TITLE
Prefilled links now work for monthly donations

### PR DESCRIPTION
**Pull request by sarthaktexas**
## Summary of the problem
We got a bug report about messages not being attached to prefilled links for monthly donations (#4700).

## Describe your changes
Fixes autofill on prefill links.

### CODE REVIEW

**Comment by sampoder** [comment]
Wouldn't it instead make sense for us to add support for messages on monthly donations? https://github.com/hackclub/hcb/pull/4590

**Comment by sarthaktexas** [comment]
> Wouldn't it instead make sense for us to add support for messages on monthly donations? #4590

Stripe doesn't support it.

**Comment by sampoder** [comment]
We don't use Stripe for the messages, we store them in our own database

File: app/controllers/donations_controller.rb
Comment (sampoder) [inline app/controllers/donations_controller.rb]:
```suggestion
        amount: params[:amount],
        message: params[:message]
```

Once we merge https://github.com/hackclub/hcb/pull/4590, we'll want to make this change.

Could you double check this for me @sarthaktexas?

Comment (sarthaktexas) [reply app/controllers/donations_controller.rb]: it'll throw an error if we commit this right now, but you'd be correct -- gary and i tried it already :)

Comment (sampoder) [reply app/controllers/donations_controller.rb]: sweet!

**Comment by garyhtou** [comment]
Let's get https://github.com/hackclub/hcb/pull/4590 cleaned up and merged, then we'll merge this one

File: app/controllers/donations_controller.rb
Comment (garyhtou) [inline app/controllers/donations_controller.rb]:
```suggestion
        amount: params[:amount],
        message: params[:message],
```

in prep for https://github.com/hackclub/hcb/pull/4590

Comment (sarthaktexas) [reply app/controllers/donations_controller.rb]: oh sam suggested this, in that case i'll commit but don't merge this pr until his is merged

Comment (garyhtou) [reply app/controllers/donations_controller.rb]: oops sorry, didn't notice!

**Comment by garyhtou** [comment]
I discarded changes to `[app/javascript/components/donation_overview/PreviewLink.js` because they were formatting changes. The same changes were made by https://github.com/hackclub/hcb/pull/4957 and it's causing a merge conflict

**Review by garyhtou — APPROVED** [review]
This is good to merge _after_ https://github.com/hackclub/hcb/pull/4590 gets merged!

**Comment by sampoder** [comment]
#4590 was merged! let's do it :D

**Pull request merged**
